Tests/TuistSupportTests/Xcode/XcodeControllerTests.swift

```diff
@@ -29,6 +29,28 @@ final class XcodeControllerTests: TuistUnitTestCase {
         XCTAssertNil(xcode)
     }
 
+    func test_selected_is_cached() throws {
```

Comment:
Beautiful!

Reply:
What about adding an integration test for this? You can add it to the `TuistSupportIntegrationTests` target. We can assume the tests are ran in a environment where Xcode exists, so we can check that calling the method doesn't error and gives you a version. We don't need to check the specific version.

That way if something changes in the bundle of Xcode, which I doubt, the test will fail.

Reply:
Added it the test case 🙂